### ISSUE TITLE
False positive color contrast on anything use pseudo code or icon

### ISSUE THREAD

**Issue by smustgrave** [dequelabs/axe-core#3431]
Product: axe-core

Expectation: Correctly identify color contrast errors

Actual: Anything that has pseudo code icons or background image fails color contrast

Motivation: Currently can't use color contrast test.

<pre><code>
axe-core version: 4.4.1

Browser and Assistive Technology versions

For Tooling issues:
- Node version: 17.6.0
- Platform:  Linux
</code></pre>

Example 
This generates errors every link with the arrow next to it.

<img width="897" alt="Screen Shot 2022-04-07 at 12 45 03 PM" src="https://user-images.githubusercontent.com/10660823/162272733-3b394271-1001-435c-8718-cfb6cfd50490.png">

**Comment by straker** [comment]
Thanks for the issue. When you say errors, is axe flagging the nodes as Violations or Needs Reviews? Axe isn't able to determine the background color of an image, so it should flag any elements on top of the image as Needs Review.

**Comment by smustgrave** [comment]
We are using the pa11y tool with axe runner

They are getting flagged as errors 

 • Elements must have sufficient color contrast (https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axeAPI)

   (#header > nav > div > ul > li:nth-child(6) > button > span)

   <span>News</span>

Out of the 6 links the 5 with the icon are flagged

**Comment by straker** [comment]
Are you able to link to the site in question or post a code sample that reproduces the error? It's hard to tell from the image, but it does look like that links with arrows are a light grey (as opposed to the Education & Outreach link which is dark bold). Those could be too light for color-contrast, but hard to say for sure.

**Comment by smustgrave** [comment]
So that screenshot is for a site that hasn't launched yet.
But another site we have https://www.pandemicoversight.gov/ the header links are flagged.  and the "We are the Pandemic Response" header is also flagged

**Comment by straker** [comment]
Thanks for the link. Running axe on the page produces 0 Violations, but does show some Needs Review items for color contrast. Looking at the elements it looks like those are expected results. Any element or ancestor which uses a pseudo element will be flag as needs review since there are no browser APIs available that tell us the width/height of the pseudo element. As such we can't determine if the element affects a background color to the element.

**Comment by smustgrave** [comment]
Using the color contrast analyzer app from DHS accessibility office and those links pass.

**Comment by smustgrave** [comment]
Ahh so they can be marked as false positives then.

**Comment by straker** [comment]
Not necessarily. We're just saying we can't with confidence say if the links pass or fail due to the presence of a pseudo element that defines a background-color. That's why we put it under Needs Review rather than pass / failure as you'll need to do a manual review of the links to make sure they pass color-contrast. 

If pa11y is saying that's a violation, then it might be an problem on their side that's marking it that way.

**Comment by smustgrave** [comment]
I'll test removing that pseudo after code and see if that fixes it

**Comment by smustgrave** [comment]
Unfortunately that wasn't it :(  Still get the error for any link that uses an icon next to it.

**Comment by dylanb** [comment]
You should have pa11y address these `incomplete` issues as items that need manual review and not as failures. We are working on improvements to the UI for the extension to handle these cases, but this is not a bug, I am going to close this issue
